guide/docs/popular-topics/permissions.mdx

```diff
@@ -1,8 +1,382 @@
+
+### Context-agnostic
+
+If you have an instance of any class implementing <DocsLink reference="disnake.abc.GuildChannel" />,
+then the easiest way to query permissions for a member/role is
+through <DocsLink reference="disnake.abc.GuildChannel.permissions_for" />.
+If you need a bit more lower level control, you can also get raw overwrite objects
+using <DocsLink reference="disnake.abc.GuildChannel.overwrites_for" />
+or <DocsLink reference="disnake.abc.GuildChannel.overwrites" />.
+
+### In prefix commands
+
+There are no direct methods on <DocsLink reference="disnake.ext.commands.Context" /> for querying permission
+information, however, you can still access <DocsLink reference="disnake.ext.commands.Context.channel" /> and
+call above-mentioned methods on it, though be sure to check that it's not <DocsLink reference="disnake.PartialMessageable" />
+as it doesn't have any permissions-related methods on it either.
+
+### In interactions
+
+When working with interactions (i.e., with application commands or components), there are two main methods
+defined on <DocsLink reference="disnake.Interaction" />: <DocsLink reference="disnake.Interaction.permissions" />
+(for getting interaction author's permissions) and <DocsLink reference="disnake.Interaction.app_permissions" />
+(for getting bot's permissions in interaction's channel).
+
+## How permissions are calculated
+
+Permissions can be a tough thing to understand at first, but understanding how permissions are
+applied can help you <!-- feel the pain of us mere mortals re-implementing discord client's obscure behaviour -->
+better grasp this concept.
+
+In case we're calculating permissions for a member, the strategy is as follows:
+
+-   If member is the guild owner, all permissions are granted - no questions asked. Otherwise..
+-   First we apply the `@everyone` role's permissions.
+-   Then, for each role the member has we apply it's permissions too.
+-   -   If at this point the member has the "Administrator" permission, all others are granted automatically
+        (the member is a guild-wide administrator, and, as such, is functionally equivalent to the owner except
+        they can't delete the guild).
+-   After that, if we're in a channel, the following is additionally done.
+-   -   We apply it's overwrites in the same order: `@everyone` overwrites,
+        role overwrites and finally member overwrites.
+-   -   Then [channel-specific implicit permissions](#channel-specific-implicit-permissions) are handled.
+-   In the end, we take into account [guild-wide implicit permissions](#guild-wide-implicit-permissions).
+
+In case of calculating permissions for a role, the strategy is the same as if we were calculating permissions
+for a member with only that role (and `@everyone`), except that member overwrites are
+not taken into account.
+
+:::note
+Curious about the implementation? You can see it [here](https://git.disnake.dev/disnake/blob/master/disnake/abc.py#L651-L808).
+:::
```

Comment:
This link will change over time, maybe point to https://discord.com/developers/docs/topics/permissions#permission-overwrites instead? :thinking:

Reply:
The note is about how disnake implements permission calculation, not how discord tells to do it. I also now remembered that we can pin link to a specific revision, which is done in 26020fb24bc0ac78072e1f577ca445723d6c41a0. Please unresolve conversation if this doesn't fit you.

Reply:
Linking to a specific ref seems good, sure.
Another approach would be instructing readers to use the `[source]` button [here](https://docs.disnake.dev/en/stable/api/abc.html#disnake.abc.GuildChannel.permissions_for), but that's sort of unnecessary here since the implementation really shouldn't be changing a whole lot.